homeassistant/core.py

```diff
@@ -333,6 +334,9 @@ def _async_stop_handler(self, *args):
     @callback
     def _async_restart_handler(self, *args):
         """Restart Home Assistant."""
+        if subprocess.call([sys.argv[0], '--script', 'check_config']):
+            _LOGGER.error("check_config failed. Not restarting.")
```

Comment:
This is a great idea, with the same notification name, you can do a "checking config" message and change it into a "failed" config message if it fails.

Checking does take a while on slower rPi's!